tokio-sync/src/mpsc/bounded.rs

```diff
@@ -111,6 +111,7 @@ pub struct RecvError(());
 /// }));
 /// ```
 pub fn channel<T>(buffer: usize) -> (Sender<T>, Receiver<T>) {
+    assert_eq!(buffer >= 1, true);
```

Comment:
How about including a message as to why the assert fails, like `assert!(buffer > 0, "bounded channel buffer size must be greater than 0")`?